From a4c0951f296c6ea1a0c2a64df3c243c436246902 Mon Sep 17 00:00:00 2001
From: Rob Caelers <rob.caelers@gmail.com>
Date: Sat, 4 May 2024 19:07:45 +0200
Subject: [PATCH] Prevent crash when wayland is disabled

---
 .../src/unix/UnixInputMonitorFactory.cc       | 21 +++++++++++--------
 1 file changed, 12 insertions(+), 9 deletions(-)

diff --git a/libs/input-monitor/src/unix/UnixInputMonitorFactory.cc b/libs/input-monitor/src/unix/UnixInputMonitorFactory.cc
index 965ea1532..7994bb8de 100644
--- a/libs/input-monitor/src/unix/UnixInputMonitorFactory.cc
+++ b/libs/input-monitor/src/unix/UnixInputMonitorFactory.cc
@@ -121,15 +121,18 @@ UnixInputMonitorFactory::create_monitor(MonitorCapability capability)
             }
 #endif
 
-          initialized = monitor->init();
-
-          if (initialized)
-            {
-              TRACE_MSG("Success");
-              break;
-            }
-
-          monitor.reset();
+          if (monitor)
+          {
+            initialized = monitor->init();
+
+            if (initialized)
+              {
+                TRACE_MSG("Success");
+                break;
+              }
+
+            monitor.reset();
+          }
 
           loop++;
           if (loop == available_monitors.end())
